### ISSUE TITLE
Use of "the current official scramble program" not always possible

### ISSUE THREAD

**Issue by timhabermaas** [thewca/wca-regulations#85]
Let's say a delegate prints out the scrambles for a competition on Thursday and TNoodle gets updated the next day*. Then the delegate would violate rule [4f](https://www.worldcubeassociation.org/regulations/#4f) if he uses these scrambles for the competition at the weekend:

>  Competition scramble sequences must be generated using the current official version of the official WCA scramble program (available via the WCA website).

Therefore Stefan proposed to update _"using the current official version"_ to _"a current official version"_ and keep the old version of TNoodle on the server for around another week.

\* Another solution would be to not update TNoodle before competitions, but this might not always be practical since there might be several competitions scattered across the next few weeks. Especially if cubing grows further in popularity.

**Comment by pochmann** [comment]
Something I just noticed: What does "current" refer to? In your example,
the scrambles _were_ generated with the then-current version.

**Comment by timhabermaas** [comment]
If "current" refers to the time the scrambles were generated will this not make any generated scrambles valid no matter when they've been generated?

Anyway, I'm not sure what it refers to either. I just assumed it means "start of the competition" or something close to it. This probably needs clarification as well.

**Comment by pochmann** [comment]
Not quite. It wouldn't allow me to use 0.7.10 now to generate scrambles.
But yes, with that interpretation I could use 0.7.12 now and use the
scrambles in two years.

I agree that something like start of the competition is probably intended,
but also think it could/should be made clearer.

**Comment by lgarron** [comment]
I'm keeping to a [pledge](https://www.worldcubeassociation.org/regulations/announcements/introducing-wrc-announcements) to release any important WRC changes Monday-Wednesday (preferring earlier in the week if there are no conflicting competitions). I would like to codify this, but I'm not sure where to put this -- the official style guide?

As for the phrasing, I'm not convinced it needs additional clarification. We're not trying to cover a case for the average/scheming competitor, but for delegates (trusted entities).
The Regulations still have lots of small details that don't really matter. The default interpretation should be that the scrambles must be generated using the scrambler that was official at the time of the competition. Delegates shouldn't be "stocking up" on scrambles too early, and we can tell from the generated files.

If a competition uses an incorrect scrambler, the Board can handle it as an incident.

I used to hear a lot of different opinions about official scrambles (the [2010 Regs](https://www.worldcubeassociation.org/regulations/history/files/regulations2010.html#scrambling) are very unclear if the provided programs are required), and I have made an effort to be very clear about transitions. This includes things like [announcing](https://www.worldcubeassociation.org/node/3110)

> Please remember that official competitions must always use the current version of the official scramble program. This is always available from: https://www.worldcubeassociation.org/regulations/scrambles/

instead of providing a direct link. This ensures that clicking old announcements will still lead to the current official scrambler.

All that said, I'm not inherently against any change. Just make sure that
1) competitions use the official scrambles
2) things are clear and easy for delegates / anyone who needs to generate scrambles.
